### PR TITLE
Unit test suit

### PR DESCRIPTION
**Pull request by SevlaMare**
-Initial setup
-Added test suit
-Linter fixes

#### Board
* board_display
Display the board at initial state

* board_update
Both players can update the board
Others players can't

#### Engine
* input_number?
Should accept numbers in range 1 to 9
only from player 1 and 2

* spot_taken?
True if spot is not taken
False if spot is taken

* win?
Return false if not have a winner
Return true if have a winner

* full_board?
Return true if the board is full
Return false if the board is not full

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# RSpec.describe Enumerable do
+#  --- VScode setup end ---
+
+describe Engine do
+  let(:game1) { Engine.new }
+  let(:field_win) { ['x', 'x', 'x', 4, 5, 6, 7, 8, 9] }
+
+  describe '#Class Board' do
+    context 'Display Board' do
+      it 'Should display the board' do
```
Comment (suretrust) [inline spec/main_spec.rb]: `[Optional]` It's best practice to start the `it` block info with a singular verb. For example:
```ruby
it 'displays the board' do
end
it 'updates the board when player 1 makes a move' do
end
it 'validates the first valid input' do
end
```

File: spec/main_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        expect(game1.board_update(1, 1))
+          .to eq('x')
+      end
+
+      it 'Player 2 can update the board' do
+        expect(game1.board_update(1, 2))
+          .to eq('o')
+      end
+    end
+  end
```
Comment (suretrust) [inline spec/main_spec.rb]: * [x] Specs should cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
For example: After testing that player 1 and 2 updates the board, also test that any random player cannot update the board.

```ruby
context 'when player 1 or 2 plays' do
      it 'updates the board' do
        expect(game1.board_update(1, 1))
          .to eq('x')
      end

      it 'updates the board' do
        expect(game1.board_update(1, 2))
          .to eq('o')
      end

      it 'does not update the board' do
        expect(game1.board_update(1, 4))
          .to eq('o')
      end
end
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @SevlaMare,

### Great job thus far! 🙌 Please see my comments on your code. 👇 

- [x] `[Optional]` You can read more about RSpec best practices [here](http://www.betterspecs.org/).

**Kindly submit another code review request once this has been fixed.** _Happy coding!_

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**